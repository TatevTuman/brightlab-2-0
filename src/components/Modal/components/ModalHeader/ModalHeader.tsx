import React, { memo } from 'react'
import { Children } from '@types'
import styles from '../../Modal.module.scss'

export interface ModalHeaderProps {
  children: Children
}

type ModalHeaderType = React.FC<ModalHeaderProps>

const ModalHeader: ModalHeaderType = props => {
  return <header className={styles.modalHeader}>{props.children}</header>
}

export default memo(ModalHeader)
