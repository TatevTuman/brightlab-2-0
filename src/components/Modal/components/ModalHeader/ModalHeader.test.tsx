import React from 'react'
import { render, cleanup, waitFor } from '@testing-library/react'
import ModalHeader from './ModalHeader'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('ModalHeader', () => {
  it('renders correctly', async () => {
    const { container, getByText } = render(<ModalHeader>children</ModalHeader>)
    const awaitedContainer = await waitFor(() => container)

    expect(awaitedContainer).toMatchSnapshot()
    expect(getByText('children')).toBeInTheDocument()
  })
})
