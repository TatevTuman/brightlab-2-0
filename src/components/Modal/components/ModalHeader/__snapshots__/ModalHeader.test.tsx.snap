// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader renders correctly 1`] = `
<div>
  <header
    class="modalHeader"
  >
    children
  </header>
</div>
`;
